pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java

```diff
@@ -425,24 +432,28 @@ private void processResult(Record srcRecord,
     }
 
     private void sendOutputMessage(Record srcRecord, Object output) {
+        Thread.currentThread().setContextClassLoader(functionClassLoader);
         try {
             this.sink.write(new SinkRecord<>(srcRecord, output));
         } catch (Exception e) {
             log.info("Encountered exception in sink write: ", e);
             stats.incrSinkExceptions(e);
             throw new RuntimeException(e);
         }
+        Thread.currentThread().setContextClassLoader(frameworkClassLoader);
```

Comment:
This should be done in a `finally {}` block, otherwise the classloader is not reset after the runtime exception is thrown.

Reply:
will fix